crates/uv/src/commands/project/sync.rs

```diff
@@ -881,3 +900,270 @@ fn store_credentials_from_target(target: InstallTarget<'_>) {
         }
     }
 }
+
+/// A report of a project's state and updates to make.
+#[derive(Debug, Serialize)]
+struct ProjectReport {
+    /// The project directory path.
+    project_dir: PathBuf,
```

Comment:
Here use `_dir`, for the env we use `_path`, we should have a shared nomenclature here (maybe `root`?)